tools/CopySchema/Program.cs

```diff
@@ -0,0 +1,202 @@
+                Environment.Exit(1);
+            }
+
+            Environment.Exit(0);
+        }
+
+        public static int CopyFiles(Dictionary<string, string> filesToCopy)
+        {
+            var count = 0;
+            foreach (var file in filesToCopy)
+            {
+                var from = file.Key;
+                var to = file.Value;
+                if (!Directory.Exists(Path.GetDirectoryName(to)))
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(to));
+                }
+
+                File.Copy(from, to);
+                count++;
+            }
+
+            return count;
+        }
+
+        public static Dictionary<string, string> GetFilesToCopy(List<KeyValuePair<PackageInfo, string>> schemaFiles,
+            string schemaRoot)
+        {
+            var results = new Dictionary<string, string>();
+            foreach (var file in schemaFiles)
+            {
+                var relativeFilePath = file.Value.Replace(Path.Combine(file.Key.Path, "Schema"), string.Empty).TrimStart(Path.DirectorySeparatorChar);
+                var newPath = Path.Combine(schemaRoot, FromGdkPackages, file.Key.Name, relativeFilePath);
+                results.Add(file.Value, newPath);
+            }
+
+            return results;
+        }
+
+        public static List<KeyValuePair<PackageInfo, string>> FindSchemaFiles(List<PackageInfo> packages)
+        {
+            var results = new List<KeyValuePair<PackageInfo, string>>();
+            foreach (var package in packages)
+            {
+                var searchPath = Path.GetFullPath(Path.Combine(package.Path, "Schema"));
+                if (!Directory.Exists(searchPath))
+                {
+                    continue;
+                }
+
+                foreach (var file in Directory.GetFiles(searchPath, "*.schema", SearchOption.AllDirectories))
+                {
+                    results.Add(new KeyValuePair<PackageInfo, string>(package, Path.GetFullPath(file)));
+                }
+            }
+
+            return results;
+        }
+
+        public static List<PackageInfo> GetPackageInfos(dynamic manifest, string packagesRoot)
+        {
+            var results = new List<PackageInfo>();
+            JObject dependencies = manifest.dependencies;
+            foreach (var dep in dependencies.Properties())
+            {
+                var value = dep.Value.ToString();
+                if (value.StartsWith("file:"))
+                {
+                    var fullPath = Path.GetFullPath(Path.Combine(packagesRoot, value.Replace("file:", string.Empty)));
+                    results.Add(new PackageInfo(dep.Name, fullPath));
+                }
+            }
+
+            return results;
+        }
+
+        public static void ParseArguments(string[] args, out string manifestFile, out string packagesRoot, out string schemaRoot)
+        {
+            if (args.Length != 2)
```

Comment:
Would it make more snse just to pull out the two array elements into two arguments? I presume this is just here so that it's easier to pass command line arguments to it.

Reply:
Especially as the two arguments have explicit names already.

Reply:
Resolved offline.